### PR TITLE
PRM-2551 pointed penalty value to postedamount when outstandingamount…

### PR DESCRIPTION
**Pull request by RoheelArshad**
… is unavailable

### CODE REVIEW

**Comment by LewisMitchell** [comment]
Ideally this needs to be rebased and not pulled from main

File: app/services/PenaltiesFrontendService.scala [hunk 1]
```diff
@@ -39,8 +39,10 @@ class PenaltiesFrontendService @Inject()() {
       _.details.map(_.map(
         oldLPPDetails => {
           (oldLPPDetails.penaltyStatus, oldLPPDetails.appealInformation) match {
-            case (_, Some(appealInformation)) if appealInformation.exists(_.appealStatus.exists(isAppealStatusUpheldOrChargeReversed)) => {
-              oldLPPDetails.copy(
+
+            case (_, Some(appealInformation)) if appealInformation.exists(_.appealStatus.contains(AppealStatusEnum.Upheld)) => {
```
Comment (LewisMitchell) [inline app/services/PenaltiesFrontendService.scala:43]: I think this is a result of merge conflicts, needs to be changed back

File: it/controllers/AppealsControllerISpec.scala [hunk 2]
```diff
@@ -234,6 +236,7 @@ class AppealsControllerISpec extends IntegrationSpecCommonBase with ETMPWiremock
       |          "penaltyCategory": "LPP1",
       |          "penaltyStatus": "P",
       |          "penaltyAmountPaid": 0,
+      |          "penaltyAmountPosted": 0,
```
Comment (LewisMitchell) [inline it/controllers/AppealsControllerISpec.scala:239]: penaltyAmountPosted = 144

File: it/controllers/AppealsControllerISpec.scala [hunk 3]
```diff
@@ -859,6 +863,7 @@ class AppealsControllerISpec extends IntegrationSpecCommonBase with ETMPWiremock
         |          "penaltyCategory": "LPP1",
         |          "penaltyStatus": "P",
         |          "penaltyAmountPaid": 0,
+        |          "penaltyAmountPosted": 0,
```
Comment (LewisMitchell) [inline it/controllers/AppealsControllerISpec.scala:866]: penaltyAmountPosted = 144

File: it/controllers/AppealsControllerISpec.scala [hunk 4]
```diff
@@ -900,6 +905,7 @@ class AppealsControllerISpec extends IntegrationSpecCommonBase with ETMPWiremock
         |          "penaltyCategory": "LPP2",
         |          "penaltyStatus": "P",
         |          "penaltyAmountPaid": 0,
+        |          "penaltyAmountPosted": 0,
```
Comment (LewisMitchell) [inline it/controllers/AppealsControllerISpec.scala:908]: penaltyAmountPosted = 144

File: it/controllers/AppealsControllerISpec.scala [hunk 5]
```diff
@@ -924,6 +930,7 @@ class AppealsControllerISpec extends IntegrationSpecCommonBase with ETMPWiremock
         |          "penaltyCategory": "LPP1",
         |          "penaltyStatus": "P",
         |          "penaltyAmountPaid": 0,
+        |          "penaltyAmountPosted": 0,
```
Comment (LewisMitchell) [inline it/controllers/AppealsControllerISpec.scala:933]: penaltyAmountPosted = 144

File: it/controllers/AppealsControllerISpec.scala [hunk 6]
```diff
@@ -965,6 +972,7 @@ class AppealsControllerISpec extends IntegrationSpecCommonBase with ETMPWiremock
         |          "penaltyCategory": "LPP2",
         |          "penaltyStatus": "P",
         |          "penaltyAmountPaid": 0,
+        |          "penaltyAmountPosted": 0,
```
Comment (LewisMitchell) [inline it/controllers/AppealsControllerISpec.scala:975]: penaltyAmountPosted = 144

File: it/controllers/AppealsControllerISpec.scala [hunk 7]
```diff
@@ -995,6 +1003,7 @@ class AppealsControllerISpec extends IntegrationSpecCommonBase with ETMPWiremock
         |          "penaltyCategory": "LPP1",
         |          "penaltyStatus": "P",
         |          "penaltyAmountPaid": 0,
+        |          "penaltyAmountPosted": 0,
```
Comment (LewisMitchell) [inline it/controllers/AppealsControllerISpec.scala:1006]: penaltyAmountPosted = 144.01

File: it/controllers/PenaltiesFrontendControllerISpec.scala [hunk 8]
```diff
@@ -482,6 +484,7 @@ class PenaltiesFrontendControllerISpec extends IntegrationSpecCommonBase with ET
         |            "principalChargeDueDate":"2022-10-30",
         |            "principalChargeBillingTo":"2022-10-30",
         |            "penaltyAmountPaid":0,
```
Comment (LewisMitchell) [inline it/controllers/PenaltiesFrontendControllerISpec.scala:486]: Could you remove penaltyAmountPaid. It wouldn't be there

File: it/controllers/PenaltiesFrontendControllerISpec.scala [hunk 9]
```diff
@@ -569,6 +572,7 @@ class PenaltiesFrontendControllerISpec extends IntegrationSpecCommonBase with ET
...
```
Comment (LewisMitchell) [inline it/controllers/PenaltiesFrontendControllerISpec.scala:574]: Could you remove penaltyAmountPaid. It wouldn't be there

File: test/controllers/PenaltiesFrontendControllerSpec.scala [hunk 10]
```diff
@@ -164,7 +164,7 @@ class PenaltiesFrontendControllerSpec extends SpecBase with LogCapturing {
               communicationsDate = Some(LocalDate.of(2022, 1, 1)),
               penaltyAmountOutstanding = Some(144),
               penaltyAmountPaid = Some(0.21),
-              penaltyAmountPosted = Some(144.21),
+              penaltyAmountPosted = 44.21,
```
Comment (LewisMitchell) [inline test/controllers/PenaltiesFrontendControllerSpec.scala:167]: 144.21

File: test/services/auditing/APIServiceSpec.scala [hunk 11]
```diff
@@ -452,7 +452,7 @@ class APIServiceSpec extends SpecBase {
                     communicationsDate = Some(LocalDate.of(2022, 1, 1)),
                     penaltyAmountOutstanding = None,
                     penaltyAmountPaid = None,
-                    penaltyAmountPosted = None,
+                    penaltyAmountPosted = 0,
```
Comment (LewisMitchell) [inline test/services/auditing/APIServiceSpec.scala:455]: Might be worth coming up with some amount. The go to is 144.21

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: it/controllers/AppealsControllerISpec.scala (hunk 4 above)
Comment (LewisMitchell) [reply it/controllers/AppealsControllerISpec.scala:908]: Still valid

File: it/controllers/APIControllerISpec.scala [hunk 12]
```diff
@@ -435,6 +439,7 @@ class APIControllerISpec extends IntegrationSpecCommonBase with ETMPWiremock wit
             |       "principalChargeMainTransaction" : "4700",
             |       "penaltyAmountOutstanding": 99.99,
             |       "penaltyAmountPaid": 1001.45,
+            |       "penaltyAmountPosted": 1101.44,
```
Comment (LewisMitchell) [inline it/controllers/APIControllerISpec.scala:442]: This would be 0

File: it/controllers/PenaltiesFrontendControllerISpec.scala [hunk 13]
```diff
@@ -481,7 +483,7 @@ class PenaltiesFrontendControllerISpec extends IntegrationSpecCommonBase with ET
         |         {
         |            "principalChargeDueDate":"2022-10-30",
         |            "principalChargeBillingTo":"2022-10-30",
-        |            "penaltyAmountPaid":0,
+        |            "penaltyAmountPosted": 99.99,
```
Comment (LewisMitchell) [inline it/controllers/PenaltiesFrontendControllerISpec.scala:486]:
```suggestion
        |            "penaltyAmountPosted": 0,
```

File: it/controllers/PenaltiesFrontendControllerISpec.scala [hunk 14]
```diff
@@ -568,7 +570,7 @@ class PenaltiesFrontendControllerISpec extends IntegrationSpecCommonBase with ET
...
```
Comment (LewisMitchell) [inline it/controllers/PenaltiesFrontendControllerISpec.scala:573]:
```suggestion
        |            "penaltyAmountPosted": 0,
```

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: it/controllers/APIControllerISpec.scala (hunk 12 above)
Comment (LewisMitchell) [reply it/controllers/APIControllerISpec.scala:442]: It's an accruing penalty :)

Comment (RoheelArshad) [reply it/controllers/APIControllerISpec.scala:442]: Ok, I'll change paid, posted and outstanding to 0

**Review by LewisMitchell — DISMISSED** [review]
Still some minor changes needed however they shouldn't affect much

File: app/models/getPenaltyDetails/latePayment/LPPDetails.scala [hunk 15]
```diff
@@ -115,10 +115,10 @@ object LPPDetails extends JsonUtils {
     }
 
     private def setPenaltyAmountOutstanding(penaltyAmountOutstanding: Option[BigDecimal])
```
Comment (LewisMitchell) [inline app/models/getPenaltyDetails/latePayment/LPPDetails.scala:117]: Remove

**Pull request merged**